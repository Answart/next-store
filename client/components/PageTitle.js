import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import Head from 'next/head';
import SvgIcon from './SvgIcon';
import { StyledPageTitle } from './styles/PageStyles';


const PageTitle = ({ page, titles }) => (
  <StyledPageTitle>
    <Head>
      <title>
        Next Store{!!page ? ` | ${page}` : ''}
      </title>
    </Head>

    {!titles || !titles.length ? (
      <span style={{ padding: '0 1rem'}}>
        {page}
      </span>
    ) : (titles.map((title, i) =>
      <span key={i}>
        {!!title.href && (
          <Link href={title.href}>
            <a className="thn-btn">
              {title.label}
            </a>
          </Link>
        )}

        {!!title.click && (
          <button className="thn-btn"
            onClick={e => title.click(e)}
          >{title.label}</button>
        )}

        {!title.href && !title.click && (
          <span style={{ padding: '0 1rem'}}>
            {title.label}
          </span>
        )}
        {(i !== (titles.length - 1)) && (
          <SvgIcon width={7} name='doubleRightArrow' />
        )}
      </span>
    ))}
  </StyledPageTitle>
);

PageTitle.defaultProps = {
  page: '',
  titles: []
};

PageTitle.propTypes = {
  page: PropTypes.string,
  titles: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    href: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
      query: PropTypes.object,
    }),
    click: PropTypes.func
  }))
};


export default PageTitle;
