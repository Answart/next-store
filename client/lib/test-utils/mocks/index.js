export {
  mockUser, mockUsers,
  mockImage, mockImageVariables,
  mockProduct, mockProducts, mockShopProductsVariables,
  mockVariant,
  mockCartItem,
  mockOrderItem, mockOrder,
} from './typeDefs';
export {
  localUserQueryEmptyCartMock, localUserQueryNoUserMock, localUserQueryCartItemMock,
  userQueryEmptyCartMock, userQueryNoUserMock, userQueryCartItemMock,
  signupMutationMock,
  signinMutationMock,
  signoutMutationMock,
  requestPasswordResetMutationMock, requestPasswordResetMutationErrorMock,
  resetPasswordMutationMock, resetPasswordMutationErrorMock,
  updatePermissionsMutationMock,
  createImageMutationMock, createImageMutationErrorMock,
  productQueryMock, productQueryErrorMock, productQueryNoProductMock, productQueryNoVariantMock,
  shopProductsQueryFilterMock,
  createProductMutationMock, createProductMutationErrorMock,
  updateProductMutationMock, updateProductMutationErrorMock,
  deleteProductMutationMock,
  createProductVariantMutationMock, createProductVariantMutationErrorMock,
  updateProductVariantMutationMock, updateProductVariantMutationErrorMock,
  deleteProductVariantMutationMock,
  addToCartMutationMock,
  updateCartItemMutationMock,
  removeFromCartMutationMock,
} from './resolvers';
