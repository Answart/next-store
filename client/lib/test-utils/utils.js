import casual from 'casual';


casual.seed(777);

const fakeUser = () => ({
  __typename: 'User',
  id: 'us3r1d',
  name: casual.name,
  email: casual.email,
  password: 'unique-password',
  permissions: ['ADMIN'],
  cart: [],
});

const fakeImage = () => ({
  __typename: 'Image',
  id: '1m4g31d',
  cloudinary_id: "111111",
  name: "peggswatch1",
  width: 22,
  height: 22,
  transformation: "",
  image_url: "peggswatch1.jpg",
  large_image_url: "peggswatch2.jpg",
});

const fakeProduct = () => ({
  __typename: 'Product',
  id: 'pr0duct1d',
  department: "accessories",
  title: "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
  description: "Limited Edition watch from the 2018 Fall fashion line.",
  category: "jewelry",
  brand: "Peggs",
  online: false,
  user: fakeUser(),
  variants: [],
  image: fakeImage(),
});

const fakeVariant = () => ({
  __typename: 'Variant',
  id: 'v4r13nt1d',
  availability: "4 in Stock!",
  color: "white",
  price: 35,
  quantity: 4,
  sale: true,
  salePrice: 30,
  size: "S",
  image: fakeImage(),
  product: fakeProduct()
});

const fakeCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'c4rt1t3m1d',
  quantity: 3,
  user: fakeUser(),
  variant: fakeVariant(),
  ...overrides,
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: '0rd3r1d',
  charge: 'ch_123',
  total: 4000,
  status: 'processing',
  quantity: 3,
  subtotal: 20,
  shipping: 1.5,
  tax: 1.02,
  total: 20,
  sales_tax_rate: 1.02,
  shipping_rate: 1.02,
  payment: 'some payment',
  items: [],
  buyer: fakeUser(),
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  updatedAt: '2018-04 - 06T19: 24: 16.000Z',
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: '0rd3r1t3m1d',
  title: 'order item title',
  price: 12,
  quantity: 2,
  color: 'green',
  size: 'L',
  image_url: 'some_img_url',
  seller: fakeUser(),
  variant: fakeVariant(),
  order: fakeOrder(),
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  updatedAt: '2018-04 - 06T19: 24: 16.000Z',
});


export {
  fakeUser,
  fakeImage,
  fakeProduct,
  fakeVariant,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
};
