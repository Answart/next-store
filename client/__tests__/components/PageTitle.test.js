import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import PageTitle from '../../components/PageTitle';


describe('<PageTitle />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <MockedProvider mocks={[]}>
        <PageTitle
          page='Edit Product'
          titles={[{
              label: 'Some Title',
              href: {
                pathname: '/product/edit',
                query: {
                  name: 'answart',
                }
              }
            }, {
              label: 'Selections',
              href: {
                pathname: '/product/selections',
                query: {
                  name: 'answart',
                }
              }
            }, {
              label: 'Add Selection'
          }]}
        />
      </MockedProvider>
    );
  });
  afterAll(() => wrapper.unmount());

  it('matches the snap shot', async () => {
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find('PageTitle'))).toMatchSnapshot();
  });

  it('renders properly', async () => {
    expect(wrapper.text()).toContain('Some TitleSelectionsAdd Selection');
  });
});
