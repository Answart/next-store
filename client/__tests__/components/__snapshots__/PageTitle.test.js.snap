// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> matches the snap shot 1`] = `
<PageTitle
  page="Edit Product"
  titles={
    Array [
      Object {
        "href": Object {
          "pathname": "/product/edit",
          "query": Object {
            "name": "answart",
          },
        },
        "label": "Some Title",
      },
      Object {
        "href": Object {
          "pathname": "/product/selections",
          "query": Object {
            "name": "answart",
          },
        },
        "label": "Selections",
      },
      Object {
        "label": "Add Selection",
      },
    ]
  }
>
  <PageStyles__StyledPageTitle>
    <div
      className="PageStyles__StyledPageTitle-sc-18r5v5c-6 fkQdsG"
    >
      <SideEffect(Head)>
        <Head />
      </SideEffect(Head)>
      <span
        key="0"
      >
        <Link
          href={
            Object {
              "pathname": "/product/edit",
              "query": Object {
                "name": "answart",
              },
            }
          }
        >
          <a
            className="thn-btn"
            href="/product/edit?name=answart"
            onClick={[Function]}
          >
            Some Title
          </a>
        </Link>
        <SvgIcon
          name="doubleRightArrow"
          width={7}
        >
          <svg
            className=""
            height="7px"
            style={Object {}}
            viewBox="0 0 284.936 284.936"
            width="7px"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <path
                d="M277.515,135.9L144.464,2.857C142.565,0.955,140.375,0,137.9,0c-2.472,0-4.659,0.955-6.562,2.857l-14.277,14.275 c-1.903,1.903-2.853,4.089-2.853,6.567c0,2.478,0.95,4.664,2.853,6.567l112.207,112.204L117.062,254.677 c-1.903,1.903-2.853,4.093-2.853,6.564c0,2.477,0.95,4.667,2.853,6.57l14.277,14.271c1.902,1.905,4.089,2.854,6.562,2.854 c2.478,0,4.665-0.951,6.563-2.854l133.051-133.044c1.902-1.902,2.851-4.093,2.851-6.567S279.417,137.807,277.515,135.9z"
                fill="#6d6c6c"
                key="0"
              />
              <path
                d="M170.732,142.471c0-2.474-0.947-4.665-2.857-6.571L34.833,2.857C32.931,0.955,30.741,0,28.267,0s-4.665,0.955-6.567,2.857 L7.426,17.133C5.52,19.036,4.57,21.222,4.57,23.7c0,2.478,0.95,4.664,2.856,6.567L119.63,142.471L7.426,254.677 c-1.906,1.903-2.856,4.093-2.856,6.564c0,2.477,0.95,4.667,2.856,6.57l14.273,14.271c1.903,1.905,4.093,2.854,6.567,2.854 s4.664-0.951,6.567-2.854l133.042-133.044C169.785,147.136,170.732,144.945,170.732,142.471z"
                fill="#6d6c6c"
                key="1"
              />
            </g>
          </svg>
        </SvgIcon>
      </span>
      <span
        key="1"
      >
        <Link
          href={
            Object {
              "pathname": "/product/selections",
              "query": Object {
                "name": "answart",
              },
            }
          }
        >
          <a
            className="thn-btn"
            href="/product/selections?name=answart"
            onClick={[Function]}
          >
            Selections
          </a>
        </Link>
        <SvgIcon
          name="doubleRightArrow"
          width={7}
        >
          <svg
            className=""
            height="7px"
            style={Object {}}
            viewBox="0 0 284.936 284.936"
            width="7px"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g>
              <path
                d="M277.515,135.9L144.464,2.857C142.565,0.955,140.375,0,137.9,0c-2.472,0-4.659,0.955-6.562,2.857l-14.277,14.275 c-1.903,1.903-2.853,4.089-2.853,6.567c0,2.478,0.95,4.664,2.853,6.567l112.207,112.204L117.062,254.677 c-1.903,1.903-2.853,4.093-2.853,6.564c0,2.477,0.95,4.667,2.853,6.57l14.277,14.271c1.902,1.905,4.089,2.854,6.562,2.854 c2.478,0,4.665-0.951,6.563-2.854l133.051-133.044c1.902-1.902,2.851-4.093,2.851-6.567S279.417,137.807,277.515,135.9z"
                fill="#6d6c6c"
                key="0"
              />
              <path
                d="M170.732,142.471c0-2.474-0.947-4.665-2.857-6.571L34.833,2.857C32.931,0.955,30.741,0,28.267,0s-4.665,0.955-6.567,2.857 L7.426,17.133C5.52,19.036,4.57,21.222,4.57,23.7c0,2.478,0.95,4.664,2.856,6.567L119.63,142.471L7.426,254.677 c-1.906,1.903-2.856,4.093-2.856,6.564c0,2.477,0.95,4.667,2.856,6.57l14.273,14.271c1.903,1.905,4.093,2.854,6.567,2.854 s4.664-0.951,6.567-2.854l133.042-133.044C169.785,147.136,170.732,144.945,170.732,142.471z"
                fill="#6d6c6c"
                key="1"
              />
            </g>
          </svg>
        </SvgIcon>
      </span>
      <span
        key="2"
      >
        <span
          style={
            Object {
              "padding": "0 1rem",
            }
          }
        >
          Add Selection
        </span>
      </span>
    </div>
  </PageStyles__StyledPageTitle>
</PageTitle>
`;
